xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/resources/ApplicationResources.properties

```diff
@@ -18,6 +18,7 @@
 # 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 # ---------------------------------------------------------------------------
 
-extension.security.analysis.advice.upgradeFromEM=This extension can be upgraded from the extension manager.
-extension.security.analysis.advice.upgradeXWiki=Please upgrade XWiki.
-extension.security.analysis.advice.upgradeEnvironment=Please upgrade your servlet engine.
\ No newline at end of file
+extension.security.analysis.advice.upgradeFromEM=Once a new version with a fix is available, this extension can be upgraded from the extension manager.
+extension.security.analysis.advice.transitive=This extension is installed because it is required by another extension. It is not advised to manually upgrade it. Instead, you should upgrade the extension requiring it. Please see them listed below.
```

Comment:
I don't understand in the picture the "globally"? Also in such case, the admin should upgrade those extensions from EM, or it can also be core extension? It's not entirely clear how Admins should handle this.

Reply:
To be confirmed by @tmortagne but I think an extension cannot be installed if it is brought by a core extension (it would become core as well).
"globally" means "installed on farm", the two terms are used in EM, not sure which one is the clearer to admins
![image](https://github.com/xwiki/xwiki-platform/assets/327856/903d2afc-e23c-41dd-bd93-5f001bb43e01)
The goal is to explain to admin that they shouldn't upgrade the listed extension, but to upgrade the extensions that are bringing it through their dependencies.

Reply:
Well the issue is that you only display "globally", you don't display "installed globally" that's why it's not clear. Now "installed on farm" might be more clear. 

> The goal is to explain to admin that they shouldn't upgrade the listed extension, but to upgrade the extensions that are bringing it through their dependencies.

Ok so they should go to EM for each of the extension you are listing in the advice to try updating them? is that correct?

Reply:
> an extension cannot be installed if it is brought by a core extension (it would become core as well).

Core extension are basically out of touch to XWiki since they are loaded by the application server (so impossible to uninstall and so impossible to upgrade them). Also, it's not allowed to install an extension in a namespace where it's already available (and core extensions are available everywhere).

Reply:
I am now using `#displayExtensionNamespace($namespace)`  so that the vocabulary stays uniform.